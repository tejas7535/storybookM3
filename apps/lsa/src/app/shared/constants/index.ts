export * from './language';
export * from './lubricant-type.enum';
export * from './lubrication-points.enum';
export * from './medium.enum';
export * from './optime.enum';
export * from './pages.enum';
export * from './power-supply.enum';
export * from './relubrication-interval.enum';
