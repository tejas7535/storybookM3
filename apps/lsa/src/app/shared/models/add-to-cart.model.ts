export interface AddToCartEventPayload {
  cart: CartItem[];
}

export interface CartItem {
  productCode: string;
  packCode?: string;
  quantity: number;
}
