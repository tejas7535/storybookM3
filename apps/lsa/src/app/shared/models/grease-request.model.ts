export interface GreaseRequest {
  id: string;
  name: string;
}
