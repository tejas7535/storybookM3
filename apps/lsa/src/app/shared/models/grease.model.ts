export interface Grease {
  title: string;
  id: string;
}
