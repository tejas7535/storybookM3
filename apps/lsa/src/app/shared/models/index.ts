export * from './add-to-cart.model';
export * from './grease.model';
export * from './page.model';
export * from './recommendation-form.model';
export * from './recommendation-form-value.model';
export * from './recommendation-request.model';
export * from './recommendation-response.model';
export * from './recommendation-table-data.model';
