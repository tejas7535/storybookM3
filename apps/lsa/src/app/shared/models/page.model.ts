import { Pages } from '../constants/pages.enum';

export interface Page {
  name: Pages;
  completed: boolean;
}
