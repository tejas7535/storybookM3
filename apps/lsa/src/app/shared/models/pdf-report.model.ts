import { ProductGroup } from '@schaeffler/pdf-generator';

export interface LubricatorTableData {
  [key: string]: string;
}

export type ProductDetails = ProductGroup;
