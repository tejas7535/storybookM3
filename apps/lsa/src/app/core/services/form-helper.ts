import { LubricantType, Medium } from '@lsa/shared/constants';
import {
  RecommendationFormValue,
  RecommendationRequest,
} from '@lsa/shared/models';

export const transformFormValue = (
  formValue: RecommendationFormValue
): RecommendationRequest => ({
  lubricationPoints: formValue.lubricationPoints.lubricationPoints,
  lubricationInterval: formValue.lubricationPoints.lubricationInterval,
  lubricationQty: formValue.lubricationPoints.lubricationQty,
  pipeLength: Number(formValue.lubricationPoints.pipeLength),
  optime: formValue.lubricationPoints.optime,
  medium:
    formValue.lubricant.lubricantType === LubricantType.Oil
      ? Medium.Oil
      : Medium.Grease,
  greaseId:
    formValue.lubricant.lubricantType === LubricantType.Arcanol
      ? formValue.lubricant.grease.id
      : undefined,
  minTemp: formValue.application.temperature.min,
  maxTemp: formValue.application.temperature.max,
  battery: formValue.application.battery,
});

export const objectCompare = <T extends object>(a: T, b: T): boolean => {
  const keysA = Object.keys(a);
  const keysB = Object.keys(b);

  if (keysA.length !== keysB.length) {
    return false;
  }

  return Object.entries(a).every(([kA, valA]) => {
    if (!Object.hasOwn(b, kA)) {
      return false;
    }

    const valB = b[kA as keyof typeof b];

    // eslint-disable-next-line unicorn/prefer-ternary
    if (
      typeof valA === 'object' &&
      typeof valB === 'object' &&
      valA !== null &&
      valB !== null
    ) {
      return objectCompare(valA, valB);
    } else {
      return valA === valB;
    }
  });
};
