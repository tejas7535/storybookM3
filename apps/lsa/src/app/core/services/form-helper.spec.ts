import {
  mockApplicationInput,
  mockLubricantInput,
  mockLubricationPointsInput,
} from '@lsa/mock-form-input';
import { RecommendationFormValue } from '@lsa/shared/models';

import { transformFormValue } from './form-helper';

describe('formHelper', () => {
  it('should transform the form value into a recommendationRequest', () => {
    const mockFormValue: RecommendationFormValue = {
      lubricationPoints: mockLubricationPointsInput,
      lubricant: mockLubricantInput,
      application: mockApplicationInput,
    };
    const result = transformFormValue(mockFormValue);

    expect(result).toMatchSnapshot();
  });
});
