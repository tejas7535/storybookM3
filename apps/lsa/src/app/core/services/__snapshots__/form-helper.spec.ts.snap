// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formHelper should transform the form value into a recommendationRequest 1`] = `
Object {
  "battery": 0,
  "greaseId": "ARCANOL_MULTI2",
  "lubricationInterval": "year",
  "lubricationPoints": "2 - 4",
  "lubricationQty": 60,
  "maxTemp": 15,
  "medium": "Grease",
  "minTemp": 5,
  "optime": 0,
  "pipeLength": 0,
}
`;
